### PR TITLE
`Communication`: Add unread message notification icon to sidebar accordion

### LINKED ISSUE #9498: Communication: Change color of chat-Icon if there are unread messages
### Is your feature request related to a problem?

In order to check in which channel I have unread messages, I have to open each channel and look manually. 

### Describe the solution you'd like

When I get a new message in a channel, e.g. direct messages, the Icon should change the color to "Artemis blue", such that I am immediately aware at a first glance.  

### Describe alternatives you've considered

The asterisk character * or red dot next to the channel Icon to indicate unread message(s). 

### Additional context

Currently the number in the round brackets only indicates the number of sub-channels: 
image

### LINKED ISSUE #9724: `Communication`: Red circle disappears even though unread messages
### Describe the bug

The red circle on the communication tab disappears even though I still have new unread messages.

### To Reproduce

0. You need an unread message
1. Go to 'the communication tab. You should see the red circle.
2. Click on the tab
3. See the channel with the unread message BUT do not click on it.
4. See error: The red circle disappears from the communication tab
Note: When you leave the communication tab (e.g. go to lecture or another course) the red circle appears again.

### Expected behavior

The red circle should stay the whole time there is an unread message. Otherwise, it isn't very clear.

### Screenshots

_No response_

### Which version of Artemis are you seeing the problem on?

7.6.5

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by asliayk**
#### General

- [X] This is a small issue that I tested locally.
- [X] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [X] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [X] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [X] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [X] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [X] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

- When a new message arrives, if the relevant accordion in the sidebar is collapsed, users cannot see which section contains the unread message without manually expanding each section.
- The red notification circle on the communication tab disappears when the user clicks on the tab, even if there are unread messages.

### Description

- Added a red notification circle to sidebar accordion sections that contain conversations with unread messages. (Closes #9498)
- Fixed the bug of red notification circle on the communication tab, it is now visible even when the user is on communication tab. (Closes #9724)
- Fixed the issue where the unread messages notification icon would shrink when the sidebar card title was long by adding a `min-width` property. 

### Steps for Testing

Prerequisites:
- 2 Instructors/Students
- 1 Course with Communication enabled

1. Log in to Artemis
2. Navigate to Communication section of a course
3. Send messages to some conversations, such as exercise/lecture channels, or direct/group messages.
4. Log in with another account—the one to which you sent a message in the previous step
5. Notice the red circle icon on the communication tab that indicates unread messages. Click on the tab and observe that the circle remains visible.
6. Notice the red circle icons next to the sidebar accordion titles, indicating that there are unread messages in the conversations within those sections.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| sidebar-accordion.component.ts | 95.12% | ✅ ❌ |

### Screenshots

**updated sidebar accordions with notification icon**
image

## Release Notes

- **New Features**
	- Introduced a notification indicator for unread messages in the sidebar accordion, enhancing user awareness of unread messages.
	- Simplified logic for displaying the `newMessage` class in the course overview component.

- **Bug Fixes**
	- Updated conditions for displaying unread message notifications to improve clarity and functionality.

- **Style**
	- Added a new CSS class for a visual indicator of unread messages in the sidebar accordion.
	- Adjusted minimum width for unread message count display to ensure consistent appearance.

- **Tests**
	- Enhanced test coverage for unread message calculations in the sidebar accordion component.

### CODE REVIEW

**Review by flbrgit — APPROVED** [review]
Tested on TS5. Everything wokrs as expected, the red dot staying until all messages in the category are read
Screenshot 2024-11-11 185907

**Comment by rabeatwork** [comment]
Could we align the styling to match the iOS app for a more consistent look and feel? Can you talk to @anian03 about this?

**Review by HawKhiem — DISMISSED** [review]
Tested on TS6. Everything works as described

image

**Review by sachmii — DISMISSED** [review]
Tested on TS4, works as described.

Bildschirmfoto 2024-11-12 um 17 31 05

File: src/main/webapp/app/shared/sidebar/sidebar-accordion/sidebar-accordion.component.ts [hunk 1]
```diff
@@ -39,6 +41,7 @@ export class SidebarAccordionComponent implements OnChanges, OnInit {
         } else {
             this.setStoredCollapseState();
         }
+        this.calculateTotalUnreadMessages();
```
Comment (SimonEntholzer) [inline src/main/webapp/app/shared/sidebar/sidebar-accordion/sidebar-accordion.component.ts:44]: isn't it kind of redundant, when you do this call in ngOnInit and ngOnChanges? Wouldn't it be enough to just do it in ngOnChanges?

File: src/main/webapp/app/shared/sidebar/sidebar-accordion/sidebar-accordion.component.ts [hunk 2]
```diff
@@ -67,6 +70,19 @@ export class SidebarAccordionComponent implements OnChanges, OnInit {
         }
     }
 
+    calculateTotalUnreadMessages(): void {
```
Comment (SimonEntholzer) [inline src/main/webapp/app/shared/sidebar/sidebar-accordion/sidebar-accordion.component.ts:73]: I think the naming here is misleading, as you do not calculate the total unread messages, but the unread messages per group.

**Review by SimonEntholzer — CHANGES_REQUESTED** [review]
Code looks good, but I have two small comments

**Comment by asliayk** [comment]
> Could we align the styling to match the iOS app for a more consistent look and feel? Can you talk to @anian03 about this?

I spoke with him and made the following updates to the design:

- Instead of a red icon, there is now a blue icon displaying the number of unread messages.
- When the sidebar accordion is expanded, the blue icon disappears, similar to the behavior in the iOS app.

I appreciate your reviews :)

**Comment by PaRangger** [comment]
Hey, I just tested on TS3 and for me the icons are not showing. I cannot find any errors in the console. I did the following steps:

Log in with 1 artemis_test_user_6 in chrome, log in with another artemis_test_user_16 in incognito chrome. With artemis_test_user_6 go to "Paul Ranggers Test Course", then go to communication and write a message, then go there with artemis_test_user_16 to see if the icon shows up. I tried to do this repeadately with logging out and in in-between but nothing seemed to make the icons show.

Bildschirmfoto 2024-11-13 um 10 26 25

**Review by PaRangger — APPROVED** [review]
Tested on TS3. Code LGTM 👍

**Review by vinceclifford — APPROVED** [review]
Tested on TS5, works as expected.

**Review by HawKhiem — APPROVED** [review]
Tested on TS5. Reapprove

**Review by sachmii — APPROVED** [review]
Tested on TS5, I like the style change. Re-approve.

**Review by krusche — APPROVED** [review]

**Pull request merged**